src/core/utils.js

```diff
@@ -393,3 +394,42 @@ export function getClientCompatibility({
 
   return { compatible, maxVersion, minVersion, reason };
 }
+
+export function getCategoryColor(category) {
+  if (!category) {
+    throw new Error('category is required.');
+  }
+
+  if (!CATEGORY_COLORS[category.type]) {
+    throw new Error(
+      `addonType "${category.type}" not found in CATEGORY_COLORS.`);
+  }
+
+  const maxColors = CATEGORY_COLORS[category.type];
+  if (category.id > maxColors) {
+    const color = parseInt(category.id / maxColors, 10);
```

Comment:
It probably would actually, I'll do that if it makes sense (which it probably does).